src/snapred/backend/recipe/algorithm/PixelDiffractionCalibration.py

```diff
@@ -156,10 +156,10 @@ def unbagGroceries(self, ingredients: Ingredients) -> None:  # noqa ARG002
                 GroupingWorkspace=self.getPropertyValue("GroupingWorkspace"),
                 DetectorPeaks=self.detectorPeaksJson,
             )
-            self.mantidSnapper.ConvertToMatrixWorkspace(
-                "Converting TOF data to MatrixWorkspace...",
-                InputWorkspace=self.wsTOF,
-                OutputWorkspace=self.wsTOF,
+            self.mantidSnapper.ConvertToEventWorkspace(
```

Comment:
What if `self.wsTOF` is a histogram workspace?

Reply:
It can't be one. The only accepted input workspace type is required to be MatrixWorkspacePropery.

Reply:
But MatrixWorkspaces can hold their data in either event or histogram format.  That's why the [`isHistogramData` method](https://docs.mantidproject.org/nightly/api/python/mantid/api/MatrixWorkspace.html#mantid.api.MatrixWorkspace.isHistogramData) exists for MatrixWorkspaces.

Reply:
I would suggest using the `HistogramValidator` with `False` to make sure this is event data, if the expectation is truly that it will only be event data.